extensions-core/lookups-cached-single/src/main/java/io/druid/server/lookup/LoadingLookup.java

```diff
@@ -62,15 +63,19 @@ public LoadingLookup(
 
 
   @Override
-  public String apply(final String key)
+  public String apply(@Nullable final String key)
   {
-    if (key == null) {
+    String keyEquivalent = NullHandling.nullToEmptyIfNeeded(key);
```

Comment:
I thought this kind of null value conversion is needed only when retrieving values from the storage layer (like in `ColumnValueSelector`), so that we expect `key` is already converted to some default value if needed. But, apparently it seems not. Would you explain why this is needed after retrieving values?

Reply:
this is to preserve backwards compatibility, 
previously, nulls and empty strings were considered same. 
and if a looking has mapping "" -> "someval" then all nulls/empty strings will be resolved to "someval".

Reply:
Thanks.